R/options.R

```diff
@@ -138,7 +138,9 @@ set_values <- function(controls, defaults, fallback_values) {
 set_select_value <- function(control, default, fallback) {
   possible_ids <- lapply(control$options, "[[", "id")
   ## Valid if not set (e.g. empty string) or it is a possible ID
-  is_valid <- function(x) !is.null(x) && (x == "" || x %in% possible_ids)
+  is_valid <- function(x) {
+    !is.na(x) && !is.null(x) && (x == "" || x %in% possible_ids)
```

Comment:
The value of `default` here this comes from a cell in a csv, so it will be length 1 but the fallback has no input validation yet (there is a big TODO for this) I've made sure to select the first of fallback before checking validity here and in the number control. Thanks for spotting this